Applications/Spire/Source/TimeAndSales/TimeAndSalesTableView.cpp

```diff
@@ -0,0 +1,273 @@
+        update_style(*money_cell, apply_table_cell_right_align_style);
+        return money_cell;
+      } else if(column_id == TimeAndSalesTableModel::Column::SIZE) {
+        auto quantity_cell = make_label(
+          to_text(table->get<Quantity>(row, column)).remove(QChar(',')));
+        update_style(*quantity_cell, apply_table_cell_right_align_style);
+        return quantity_cell;
+      } else if(column_id == TimeAndSalesTableModel::Column::MARKET) {
+        return make_label(
+          QString::fromStdString(table->get<std::string>(row, column)));
+      } else if(column_id == TimeAndSalesTableModel::Column::CONDITION) {
+        return make_label(
+          to_text(table->get<TimeAndSale::Condition>(row, column)));
+      }
+      return new QWidget();
+    }();
+    auto indicator = time_and_sales->get_bbo_indicator(row);
+    if(indicator == BboIndicator::UNKNOWN) {
+      match(*cell, UnknownIndicator());
+    } else if(indicator == BboIndicator::ABOVE_ASK) {
+      match(*cell, AboveAskIndicator());
+    } else if(indicator == BboIndicator::AT_ASK) {
+      match(*cell, AtAskIndicator());
+    } else if(indicator == BboIndicator::INSIDE) {
+      match(*cell, InsideIndicator());
+    } else if(indicator == BboIndicator::AT_BID) {
+      match(*cell, AtBidIndicator());
+    } else if(indicator == BboIndicator::BELOW_BID) {
+      match(*cell, BelowBidIndicator());
+    }
+    return cell;
+  }
+}
+
+TableView* Spire::make_time_and_sales_table_view(
+    std::shared_ptr<TimeAndSalesTableModel> table, QWidget* parent) {
+  auto table_view = TableViewBuilder(table).
+    set_header(make_header_model()).
+    set_item_builder(std::bind_front(&item_builder, table)).make();
+  update_style(*table_view, apply_table_view_style);
+  auto& header = table_view->get_header();
+  auto properties = make_header_item_properties();
+  auto table_header_menu = new ContextMenu(header);
+  for(auto i = 0; i < std::ssize(properties); ++i) {
+    header.get_widths()->set(i, properties[i].m_width);
+    auto item = header.get_item(i);
+    item->setVisible(properties[i].m_is_visible);
+    update_style(*item, [&] (auto& style) {
+      style.get(Any() > TableHeaderItem::Label()).
+        set(TextAlign(properties[i].m_alignment));
+    });
+    auto model =
+      table_header_menu->add_check_box(item->get_model()->get().m_name);
+    model->set(properties[i].m_is_visible);
+    model->connect_update_signal([=] (auto checked) {
+      table_view->get_header().get_item(i)->setVisible(checked);
+    });
+  }
+  header.setContextMenuPolicy(Qt::CustomContextMenu);
+  QObject::connect(&header, &QWidget::customContextMenuRequested,
+    [=] (const auto& pos) {
+      table_header_menu->window()->move(
+        table_view->get_header().mapToGlobal(pos));
+      table_header_menu->window()->show();
+    });
+  auto body = &table_view->get_body();
+  auto pull_indicator = make_pull_indicator(body);
```

Comment:
This functionality is good as well, factor it out into its own function within this .cpp file/translation unit within an anonymous namespace.

That way ```make_time_and_sales_table_view``` can be broken down into two separate parts, one that creates the context menu and one that adds the spinner feature, and these parts can be viewed/read separately.